### DISCUSSION TITLE
Example of useHover's handleClose function

### DISCUSSION THREAD

**Discussion by section14** [floating-ui/floating-ui#2802] [React] [answered]
Is there a boiler plate example for `customFunction()` in the example below?

```js
const hover = useHover(context, { handleClose: customFunction() })
```

I took a look at the source code for `safePolygon`, but I'm not completely sure what I should be doing. It looks like I could return something like this to keep a floating window open, but I'm getting errors:

```js
const customFunction = () => {
    return { __options: { blockPointerEvents: false } }
}
```

---

My goal is to hover over an element, and keep the floating window open until a user clicks anywhere but the window. Or, closes it from the floating window itself (that's already implemented).

The function above actually achieves my goal (not sure how), but I get the following error

```
Uncaught TypeError: handleCloseRef.current is not a function at HTMLDivElement.onScrollMouseLeave
``` 

So, I'm clearly missing something. Any ideas?

Thanks!

**Comment by atomiks** [comment, marked as answer]
Yeah although it technically works, the assumption is that `handleClose` is only to be used with `safePolygon` right now due to the `__options`; but here's how to use it:

```js
function createHandler(context) {
  return function onMouseMove(event) {
    // ...
  }
}
// This is an implementation detail, and _should_ be optional
createHandler.__options = { blockPointerEvents: false };
```

```js
useHover(context, {
  handleClose: createHandler
});
```

**Comment by silverwind** [comment]
Got this to work as well after adding `__options` and conditionally calling `onClose`.

`handleClose` is definitely lacking docs on how to use it and this "function with props" should be removed as well ideally, these are just weird to use.

**Reply by atomiks** [reply]
Yeah I agree, as I mentioned it's not really supposed to be used with anything other than `safePolygon` right now, however, this can be changed of course to make the API better public-facing/documented.

**Reply by silverwind** [reply]
FYI, I'm using `handleClose` to implement my own "safe area" around tooltips, basically the `interactive` option of tippy.js. Not sure if that'd have been possible with `safePolygon`.

<img width="178" alt="image" src="https://github.com/floating-ui/floating-ui/assets/115237/2827a9f6-4877-421d-8d23-2a6864bcc27a">
